core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -512,10 +516,38 @@ public void kill() throws InterruptedException {
             }
             
             LOGGER.log(FINER, "Killing {0}", getPid());
+            // Firstly try to kill it gracefully, then do a forcekill if it does not help (algorithm is described in JENKINS-17116)
+            killSoftly();
             p.kill();
             killByKiller();
         }
 
+        private void killSoftly() {
+            // send Ctrl+C to the process
+            try {
+                if (!p.sendCtrlC()) {
+                    return;
+                }
+            }
+            catch (WinpException e) {
+                return;
```

Comment:
I would log it, at least on the FINE level

Reply:
Ok!